src/rgw/rgw_civetweb_frontend.cc

```diff
@@ -64,6 +64,9 @@ int RGWCivetWebFrontend::run()
     conf_map["run_as_user"] = std::move(uid_string);
   }
 
+  /* Set authentication domains, this is needed for handling absolute uris in
+     requests */
+  conf_map["authentication_domain"] = g_conf->rgw_dns_name;
```

Comment:
currently the problem is that civetweb validates that the request is made to the same domain, or errors out there we could probably modify civetweb to not do any checking at all on domain and only rely on our checks,